### PR TITLE
Update simple builder for marketplace

### PR DESCRIPTION
**Pull request by ss-es**
### This PR: 
Updates the simple builder to produce bundles for the marketplace

### CODE REVIEW

File: crates/example-types/src/node_types.rs
Comment (elliedavidson) [inline crates/example-types/src/node_types.rs:140]: It would be nice to test the upgrade from v2 to v3 here :)

**Review by elliedavidson — COMMENTED** [review]
LGTM!  Just one minor comment.

File: crates/task-impls/src/builder.rs
Comment (QuentinI) [inline crates/task-impls/src/builder.rs:81]: This will make Client incompatible with legacy builders. Real marketplace builder also mounts its API on `block_info` actually, so that too.
Is it possible to mount two APIs on the same path in SimpleBuilderSource?

**Review by QuentinI — DISMISSED** [review]

File: crates/example-types/src/node_types.rs
Comment (ss-es) [reply crates/example-types/src/node_types.rs:140]: This was just for testing -- I wanted to make it easy to run the examples off the branch. I'll be reverting this before merging, and relying on integration tests (rather than the examples)

File: crates/task-impls/src/builder.rs
Comment (ss-es) [reply crates/task-impls/src/builder.rs:81]: Yeah, this is the part I need to figure out how to do properly -- we can't put two on the same endpoint in a single builder.

But if the real builder is serving at `block_info`, maybe it would be better to just have multiple simple builders -- one for each version? I'll look at this again and continue on Zulip

File: crates/testing/src/block_builder/mod.rs
Comment (ss-es) [reply crates/testing/src/block_builder/mod.rs]: This is just for the random builder -- I'd have to implement `v0_3::data_source::BuilderDataSource` for `RandomBuilderSource`, but didn't think we cared about the random builder (but I can try to do that if we want it)

File: crates/testing/src/block_builder/simple.rs
Comment (ss-es) [reply crates/testing/src/block_builder/simple.rs:194]: Done!

File: crates/task-impls/src/builder.rs
Comment (ss-es) [reply crates/task-impls/src/builder.rs]: Updated: we now have multiple clients at different module paths, which are defined as constants in `types/src/constants.rs`

File: crates/examples/infra/mod.rs
Comment (ss-es) [reply crates/examples/infra/mod.rs:410]: I see, yes that makes sense! I'll take a look at this and try to get a follow-up PR by today

**Review by elliedavidson — APPROVED** [review]

**Comment by elliedavidson** [comment]
Do we know why one of the tests failed?

**Comment by ss-es** [comment]
> Do we know why one of the tests failed?

Oh good catch, I thought it was just flaky and reran it but forgot to look at the result! I'll fix it before merging

**Review by elliedavidson — COMMENTED** [review]
This looks good to merge, but I see there is still a test failing.  I leave it up to @bfish713 whether it is okay to merge with this failure or not.  (It doesn't seem like the failure is related to these code changes).

**Pull request merged**
